routes/rooms.go

```diff
@@ -56,6 +74,77 @@ func GetEmergencyRoomById(ctx *gin.Context) {
 	ctx.JSON(http.StatusOK, resp)
 }
 
+// GetEmergencyRooms godoc
+// @Summary    get all emergency rooms
+// @Tags       emergency-rooms
+// @Produce    json
+// @Param      page                            query       uint                    false   "0-indexed page number, 0 is assumed when omitted"
+// @Success    200                             {object}    GetMultipleERsResponse
+// @Failure    501                             {object}    HTTPErrorResponse
+// @Router     /er                             [get]
+func GetEmergencyRooms(ctx *gin.Context) {
+	log, logCtx := logging.GetRequestLogger(ctx)
+	db := models.GetDB(logCtx)
+
+	erPageRaw := ctx.Param("page")
+	log.Debug().Str("getting all emergency rooms on page", erPageRaw).Send()
+
+	page, err := strconv.ParseUint(erPageRaw, 10, 32)
+
+	if err != nil {
+		SendError(ctx, http.StatusBadRequest, errors.New("invalid page number"))
+		return
+	}
+
+	var er []models.EmergencyRoom
+	tx := db.Find(&er)
+
+	const pageSize = 10
+	erRooms := make([]GetSingleERResponse, 0)
+
+	var numPages = math.Ceil(float64(len(er)) / float64(pageSize))
+	var lastPage = false
+
+	row, err := tx.Scopes(newPaginate(pageSize, int(page+1)).paginatedResult).Find(&er).Rows()
+
+	var erRoom models.EmergencyRoom
+
+	if page+1 == uint64(numPages) {
+		lastPage = true
+	}
+
+	for row.Next() {
+		err := row.Scan(&erRoom.ID, &erRoom.Name, &erRoom.Location,
```

Comment:
This brings up more questions. Why do we need to scan? The docs only use it together with Raw Queries, gorm should be able to figure out what it's operating on or we are doing something wrong.

Did you have a look at this? https://gorm.io/docs/scopes.html#Pagination